### PR TITLE
Ret 4169

### PR DESCRIPTION
**Pull request by hsjhita**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RET-4169
https://tools.hmcts.net/jira/browse/RET-4170

### Change description ###

Added new event for ACAS Cert lookup
Added ACAS Cert regex to fields
Made feeGroupReference optional

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/AcasCertificateController.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    private static final String GENERATED_DOCUMENT_URL = "Please download the ACAS Certificate from : ";
+    private final VerifyTokenService verifyTokenService;
+    private final AcasService acasService;
+
+    public AcasCertificateController(VerifyTokenService verifyTokenService, AcasService acasService) {
+        this.verifyTokenService = verifyTokenService;
+        this.acasService = acasService;
+    }
+
+    @PostMapping(value = "/retrieveCertificate", consumes = APPLICATION_JSON_VALUE)
```
Comment (GauravChawlaMOJ) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/AcasCertificateController.java:42]: May be add log when this callback is called? I know we can argue against it as well but it is helpful in future investigating any issues we may have.

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/AcasService.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        this.acasApiUrl = acasApiUrl;
+        this.acasApiKey = acasApiKey;
+    }
+
+    public List<String> getAcasCertificate(CaseData caseData, String authToken) throws JsonProcessingException {
+        if (isNullOrEmpty(caseData.getAcasCertificate())) {
+            return List.of("ACAS Certificate cannot be null or empty");
+        }
+
+        List<String> acasCertificates = List.of(caseData.getAcasCertificate());
```
Comment (GauravChawlaMOJ) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/AcasService.java:56]: if acasCertificate is a string(just 1 cert?)then why do we need a list here?

Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/AcasService.java:56]: The API from ACAS takes in multiple arguments hence why I put it into a List

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/AcasCertificateController.java (hunk 1 above)
Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/AcasCertificateController.java:42]: Log messages for callbacks added

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/AcasService.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        Object acasCertificateObject;
+        try {
+            acasCertificateObject = fetchAcasCertificates(acasCertificates.toArray(new String[0])).getBody();
+            if (ObjectUtils.isEmpty(acasCertificateObject)) {
+                return List.of("Error reading ACAS Certificate");
+            }
+        } catch (HttpClientErrorException errorException) {
+            return List.of("Error retrieving ACAS Certificate");
```
Comment (GauravChawlaMOJ) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/AcasService.java:64]: Can we either log or show (preferably just log) HttpClientErrorException if it happens

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/AcasService.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        }
+
+        AcasCertificate acasCertificate = convertAcasResponse((ArrayList) acasCertificateObject);
+        if (NOT_FOUND.equals(acasCertificate.getCertificateDocument())) {
+            return List.of("No ACAS Certificate found");
+        }
+
+        DocumentInfo documentInfo = convertCertificateToPdf(caseData, acasCertificate, authToken);
+        if (ObjectUtils.isEmpty(documentInfo)) {
+            return List.of("Error uploading ACAS Certificate");
```
Comment (GauravChawlaMOJ) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/AcasService.java:74]: May be add a try/catch block to convertCertificateToPdf and log any errors?

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/AcasService.java (hunk 3 above)
Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/AcasService.java:64]: Exception logs

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/AcasService.java (hunk 4 above)
Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/AcasService.java:74]: Surrounded in try catch. Some of the methods called from this also are surrounded in try catch statements for example uploading docs

**Review by GauravChawlaMOJ — APPROVED** [review]

**Review by furnivall — APPROVED** [review]

**Review by GauravChawlaMOJ — APPROVED** [review]

**Pull request merged**